### PR TITLE
Add more detailed documentation on the different types of request

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**
(Based on #192.)

### CODE REVIEW

**Review by nemanja-boric-sociomantic — DISMISSED** [review]

**Comment by nemanja-boric-sociomantic** [comment]
Sorry, wrong one.

**Comment by gavin-norman-sociomantic** [comment]
#192 merged. Unblocked.

File: src/swarm/neo/client/RequestHandlers.d [hunk 1]
```diff
@@ -5,6 +5,27 @@
...
+           We have encountered three sub-types of all-nodes requests, so far:
+            a. Short-lived, non-suspendable, getting requests. The nodes sends a
+               stream of messages, followed by a "finished" message.
+            b. Batch-based, suspendable requests. The node sends a batch of
+               records in a single message, then the client sends a response
+               telling the node to either send another batch or to end the
+               request. The request can be instantaneously suspended by the user
+               by halting iteration of the received batch in memory.
+            c. One-shot commands. The client sends a command to all nodes and
+               the nodes respond with a status code.
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/neo/client/RequestHandlers.d]: What about `Mirror`-like requests - long lived, non-suspendable (?), getting requests?

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/client/RequestHandlers.d]: Mirror is suspendable, at least currently. Though it's not batch-based. It's currently undetermined which direction it'll go in:
1. A batch-based request (that could be suspendable).
2. A non-suspendable, stream-based request.
3. Something else?

For now, I could add a fourth type (suspendable, streaming) to this list.

Comment (nemanja-boric-sociomantic) [reply src/swarm/neo/client/RequestHandlers.d]: > For now, I could add a fourth type (suspendable, streaming) to this list.

I feel that would be enough.

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/client/RequestHandlers.d]: Updated.

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Pull request merged**